File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        int r=0;
        while(r<nums.size()){
            if(mp.find(nums[r])!=mp.end()){
                int length=r-mp[nums[r]];
                if(length<=k)
                  return true;
            }
            mp[nums[r]]=r;
            r++;
        }
        return false;
    }
};

File: Data Structures & Algorithms/maximum-product-difference-between-two-pairs/submission-2.py
class Solution:
    def maxProductDifference(self, nums: List[int]) -> int:
        
        maxi=float('-inf')
        secmaxi=float('-inf')
        mini=float('inf')
        secmini=float('inf')

        for i in range(len(nums)):
            if nums[i]>maxi:
                secmaxi=maxi
                maxi=nums[i]
                
            elif nums[i]>secmaxi:
                secmaxi=nums[i]
    

            if nums[i]<mini:
                secmini=mini
                mini=nums[i]  
                b=mini*secmini
            elif nums[i]<secmini:
                secmini=nums[i]
    
           
        a=maxi*secmaxi  
        b=mini*secmini  

        return a-b          